validate/parsed.js

```diff
@@ -191,8 +191,12 @@ module.exports = function(style) {
         for (var k in val) {
             var speckey = k.split('.')[0]; // treat 'paint.*' as 'paint'
             var def = spec[speckey] || spec['*'];
+            var keyParts = speckey.split('-');
+
             if (def) {
                 (validators[speckey] || validate)((key ? key + '.' : key) + k, val[k], def);
+            } else if (keyParts.pop() === 'transition' && spec[keyParts.join('-')].transition) {
```

Comment:
I'm not comfortable with the side-effecting use of `pop` here, combined with `join` in the same expression. A regex would be clearer:

```js
var transition = speckey.match(/^(.*)-transition$/);
...
} else if (transition && spec[transition[1]].transition) }
...
```

Please also add tests for the following cases:

* gives an appropriate error given a transition for a property that doesn't support transitions
* gives an appropriate error given a transition for a non-existent property